### PR TITLE
Terraform 7459 redis AUTH support in beta

### PR DESCRIPTION
**Pull request by melinath**
Fixes https://github.com/hashicorp/terraform-provider-google/issues/7459.

Adds support for redis AUTH in beta.

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
redis: Added `auth_enabled` field to `google_redis_instance` (google-beta provider only)
```

### CODE REVIEW

**Comment by melinath** [comment]
I'm not quite clear on whether I need to make any changes to the redis data source as part of this. I don't see an obvious way to update the data source, but the data source docs also don't seem to be getting automatically updated to include auth_enabled.

File: third_party/terraform/tests/resource_redis_instance_test.go.erb [hunk 1]
```diff
@@ -69,6 +70,37 @@ func TestAccRedisInstance_regionFromLocation(t *testing.T) {
...
+	vcrTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		ExternalProviders: map[string]resource.ExternalProvider{
+			"random": {},
+		},
+		CheckDestroy: testAccCheckRedisInstanceDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccRedisInstance_redisInstanceAuthEnabled(context),
```
Comment (rileykarson) [inline third_party/terraform/tests/resource_redis_instance_test.go.erb:91]: Do you mind adding a second update step, which sets `auth_enabled` to false? Generally tests that update to the zero value for a type tend to catch a lot of weird behaviour in APIs.

**Review by rileykarson — COMMENTED** [review]
As written, the Redis datasource will automatically pick up the new field! Check out https://www.terraform.io/docs/providers/google/d/container_cluster.html for how we structure most data source docs these days- you could change the redis one to look like that too, and get rid of inconsistencies between the resource/ds altogether.

**Comment by melinath** [comment]
Oh, I was assuming the data source docs were auto-generated from the data source 😂  that explains it! I've adjusted the docs; I can add more or different attributes if that would be better? And I've made the test also update to disable auth.

File: third_party/terraform/tests/resource_redis_instance_test.go.erb (hunk 1 above)
Comment (melinath) [reply third_party/terraform/tests/resource_redis_instance_test.go.erb:91]: done!

File: third_party/terraform/website/docs/d/redis_instance.html.markdown [hunk 2]
```diff
@@ -9,16 +9,26 @@ description: |-
 
 # google\_redis\_instance
 
-Get information about a Google Cloud Redis instance. For more information see
-the [official documentation](https://cloud.google.com/memorystore/docs/redis)
-and [API](https://cloud.google.com/memorystore/docs/redis/apis).
+Get info about a Google Cloud Redis instance from its name.
```
Comment (rileykarson) [inline third_party/terraform/website/docs/d/redis_instance.html.markdown:12]:
```suggestion
Get info about a Google Cloud Redis instance.
```

No need to call out the required attributes imo (I didn't notice the example I linked did that)

**Review by rileykarson — APPROVED** [review]

**Comment by nitishkrishna** [comment]
Hey @melinath 
If we enable auth for a redis instance via this provider, is is possible to generate the auth string via TF as well?
Or do we still need to use the gcloud command to get that?

**Comment by melinath** [comment]
@nitishkrishna I think that might be possible; could you [open a new `terraform-google` issue](https://github.com/hashicorp/terraform-provider-google/issues/new/choose) with details on your use case & a potential terraform configuration?

**Comment by nitishkrishna** [comment]
Hey @melinath :
I have filed an issue here - https://github.com/hashicorp/terraform-provider-google/issues/7601

**Pull request merged**